hg-fast-export.py

```diff
@@ -31,6 +33,10 @@
 # write some progress message every this many file contents written
 cfg_export_boundary=1000
 
+# those 2 dict should be pickled for future resume
+attribute_tree = {}
+attribute_file = {}
+
```

Comment:
Why not load the information you need from git in export_commit() and then and then pass the info down to export_file_contents() where it is updated and then finish export_commit() by updating the relevant .gitattributes files?